pkg/controller/workspace/provision/deployment.go

```diff
@@ -203,6 +200,13 @@ func getSpecDeployment(
 		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, getPersistentVolumeClaim())
 	}
 
+	workspaceCreator := workspace.Annotations[config.WorkspaceCreatorAnnotation]
+	if workspaceCreator != "" {
```

Comment:
1. It means that kube:admin is not able to create workspaces anymore if webhooks are enabled since validation webhooks won't allow deployment creation without creator annotation.
![Screenshot_20200514_105951](https://user-images.githubusercontent.com/5887312/81908606-1c60d400-95d2-11ea-903d-0188c6c6518a.png)
2. There is one more place - podTemplate which probably should be changed as well.

Probably it should be changed in the following way:

```go
workspaceCreator, present := workspace.Annotations[config.WorkspaceCreatorAnnotation]
	if present {
		deployment.Annotations = map[string]string{
			config.WorkspaceCreatorAnnotation: workspaceCreator,
		}
		deployment.Spec.Template.Annotations = map[string]string{
			config.WorkspaceCreatorAnnotation: workspaceCreator,
		}
	}
```
but it rewrites whole annotations... It may be a bit safer if we initialize annotations above with empty map and then here put needed value

Reply:
Updated, thanks for the suggestion, my initial PR was a bit sloppy. I've rebased on master and included the Pod label as well.

Reply:
Ah I had to rewrite this portion and that check got lost in the rebase. I'll fix.

Reply:
Ah I had to rewrite this portion and that check got lost in the rebase. Should be fixed now.